online-shop/src/app/state/actions/product.actions.ts

```diff
@@ -0,0 +1,112 @@
+import { createAction, props } from '@ngrx/store';
+import {ProductDisplay} from "../../interfaces/ProductDisplay";
+import {Product} from "../../interfaces/Product";
+import {ProductAdd} from "../../interfaces/ProductAdd";
+import {ProductInCart} from "../../interfaces/ProductInCart";
+import {ProductCheckoutData} from "../../interfaces/ProductCheckoutData";
+
+
+//show products
+export const loadProducts = createAction('[Show Products Page] Load Products')
+
+export const loadProductsSuccess = createAction(
+  '[Product API] Products Load Success',
+  props<{ products: ProductDisplay[] }>()
+);
+
+export const loadProductsFailure = createAction(
```

Comment:
I'd not make a failure action for each. You don't really need more than 1 for the entire state.